api/src/services/export-services/export-service.ts

```diff
@@ -0,0 +1,145 @@
+import { PassThrough } from 'stream';
+import { IDBConnection } from '../../database/db';
+import { getS3SignedURLs, uploadStreamToS3 } from '../../utils/file-utils';
+import { getLogger } from '../../utils/logger';
+import { DBService } from '../db-service';
+import { ExportConfig } from './export-strategy';
+import { getArchiveStream, getJsonStringifyTransformStream, getQueryStream, handleStreamEvents } from './export-utils';
+
+const defaultLog = getLogger('/api/src/services/export-services/export-service.ts');
+
+const EXPORT_ARCHIVE_MIME_TYPE = 'application/zip';
+
+/**
+ * Provides functionality for exporting data.
+ *
+ * @export
+ * @class ExportService
+ * @extends {DBService}
+ */
+export class ExportService extends DBService {
+  constructor(connection: IDBConnection) {
+    super(connection);
+  }
+
+  /**
+   * Export the results of a set of SQL queries to S3.
+   *
+   * @param {ExportDataConfig} config
+   * @return {*}  {Promise<string[]>}
+   * @memberof ExportService
+   */
+  async export(exportConfig: ExportConfig): Promise<string[]> {
+    defaultLog.debug({ label: 'export', message: 'exportConfig', exportConfig });
+
+    if (exportConfig.exportStrategies.length === 0) {
+      throw new Error('No export strategies have been.');
+    }
+
+    const dbClient = await this.connection.getClient();
+
+    try {
+      const archiveStream = getArchiveStream();
+
+      handleStreamEvents(archiveStream);
+
+      const s3UploadStream = new PassThrough();
+
+      handleStreamEvents(s3UploadStream);
+
+      // Pipe the archive stream to the s3 upload stream
+      archiveStream.pipe(s3UploadStream);
+
+      // Start the S3 upload
+      const uploadPromise = uploadStreamToS3(s3UploadStream, EXPORT_ARCHIVE_MIME_TYPE, exportConfig.s3Key);
+
+      await Promise.all(
+        exportConfig.exportStrategies.map(async (exportStrategy) => {
+          const exportStrategyConfig = await exportStrategy.getExportStrategyConfig(this.connection);
+
+          defaultLog.silly({ label: 'export', message: 'exportStrategyConfig', exportStrategyConfig });
```

Comment:
Is silly intentional here?

Reply:
Yeah, though its questionable if its needed. I could probably remove it as it probably doesnt add any real value.

Reply:
Its easy enough to know what the `exportStrategyConfig` is, so logging it probably doesnt help

Reply:
removed